File: Data Structures & Algorithms/concatenation-of-array/submission-0.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        arr = [None] * 2 * len(nums)
        for i in range(len(nums)):
            arr[i] = nums[i]
            arr[i + len(nums)] = nums[i]
        return arr    
            

File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {
    private List<List<Integer>> ans;
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        this.ans = new ArrayList<>();
        helper(new ArrayList<>(), target, 0, nums);
        return ans;
    }

    private void helper(List<Integer> curr, int target, int i, int[] nums)
    {
        if(i >= nums.length)
            return;
        if(target == 0)
        {
            ans.add(new ArrayList<>(curr));
            return;
        }
        if(target < 0)
            return;
        curr.add(nums[i]);
        helper(curr, target-nums[i], i, nums);    

        curr.remove(curr.size()-1);
        helper(curr, target, i+1,nums);
    }
}
